File: Examples/VisualBasic/Outlook/ExposeProperties.vb
Imports Aspose.Email.Mail

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Email for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Email for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Email.Examples.VisualBasic.Email.Outlook
    Class Exposeproperties
        Public Shared Sub Run()
            Dim dataDir As String = RunExamples.GetDataDir_Outlook()

            ' Load mail message
            Dim msg As MailMessage = MailMessage.Load(dataDir & Convert.ToString("Message.eml"), New EmlLoadOptions())

            ' Subject
            If msg.Subject IsNot Nothing Then
                Console.WriteLine(msg.Subject)
            Else
                Console.WriteLine("no subject")
            End If

            ' From
            If msg.From IsNot Nothing Then
                Console.WriteLine(msg.From)
            Else
                Console.WriteLine("No sender")
            End If

            ' To
            If msg.[To] IsNot Nothing Then
                Console.WriteLine(msg.[To])
            Else
                Console.WriteLine("No one in To")
            End If

            ' Cc
            If msg.CC IsNot Nothing Then
                Console.WriteLine(msg.CC)
            Else
                Console.WriteLine("No one in cc")
            End If
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Outlook/ExposeProperties.cs
using System;
using Aspose.Email.Mapi;
using Aspose.Email.Mime;

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Email for .NET API reference 
when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Email for .NET API from http://www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.Email.Examples.CSharp.Email.Outlook
{
    class ExposeProperties
    {
        public static void Run()
        {
             string dataDir = RunExamples.GetDataDir_Outlook();

            // Load mail message
            MailMessage msg = MailMessage.Load(dataDir + "Message.eml", new EmlLoadOptions());
          
            // Subject
            if (msg.Subject != null)
                Console.WriteLine(msg.Subject);
            else
                Console.WriteLine("no subject");

               // From
            if (msg.From != null)
                Console.WriteLine(msg.From);
            else
                Console.WriteLine("No sender");

            // To
            if (msg.To != null)
                Console.WriteLine(msg.To);
            else
                Console.WriteLine("No one in To");

            // Cc
            if (msg.CC != null)
                Console.WriteLine(msg.CC);
            else
                Console.WriteLine("No one in cc");
        }
    }
}